tls/x509utils/certpool/list.go

```diff
@@ -0,0 +1,111 @@
+package certpool
+
+import "container/list"
+
+// List is a typed wrapper on top of [list.List]
+type List[T comparable] list.List
+
+// Sys returns the native [list.List]
+func (l *List[_]) Sys() *list.List {
+	if l == nil {
+		return nil
+	}
+	return (*list.List)(l)
+}
+
+// Zero returns the zero value of the type associated
+// to the list
+func (*List[T]) Zero() T {
+	var zero T
+	return zero
+}
+
+// PushBack adds a non-zero value at the end of the list
+func (l *List[T]) PushBack(v T) {
+	if v != l.Zero() {
+		l.Sys().PushBack(v)
+	}
+}
```

Comment:
@karasz ----^ ?

Reply:
I am more interested in the fact that `l.Sys()` can produce a `nil` value and then all there Methods will panic. I am thorn between the surprise effect that you cannot add a `zero` value to the end of the list and the fact that I do not know what was the intended purpose. Also since `list.PushBack(v)` returns a pointer to the newly added `Element`, I would like that treated here even if like:
```go

func (l *List[T]) PushBack(v T) {
    if ll := l.Sys(); ll!=nil {
        _ = ll.PushBack(v)
    }
}
```

Reply:
wouldn't it be "natural" and expected to get a panic when calling a method that doesn't return errors over a nil receiver? @karasz

Reply:
```go
if ll == nil } {
   panic(core.ErrNilReceiver)
}
``` ?